src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilder.cs

```diff
@@ -6,68 +6,54 @@
 using System.ComponentModel;
 using Microsoft.AspNetCore.Connections;
 using Microsoft.AspNetCore.SignalR.Internal.Protocol;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public class HubConnectionBuilder : IHubConnectionBuilder
     {
-        private readonly Dictionary<KeyValuePair<string, Type>, object> _settings = new Dictionary<KeyValuePair<string, Type>, object>();
-        private Func<IConnection> _connectionFactoryDelegate;
-
-        public void ConfigureConnectionFactory(Func<IConnection> connectionFactoryDelegate) =>
-            _connectionFactoryDelegate = connectionFactoryDelegate;
-
-        public void AddSetting<T>(string name, T value)
-        {
-            _settings[new KeyValuePair<string, Type>(name, typeof(T))] = value;
-        }
-
-        public bool TryGetSetting<T>(string name, out T value)
-        {
-            value = default;
-            if (!_settings.TryGetValue(new KeyValuePair<string, Type>(name, typeof(T)), out var setting))
-            {
-                return false;
-            }
-
-            value = (T)setting;
-            return true;
-        }
+        public IServiceCollection Services { get; } = new ServiceCollection();
 
         public HubConnection Build()
         {
-            if (_connectionFactoryDelegate == null)
+            var provider = Services.BuildServiceProvider();
+
+            var connectionFactory = provider.GetService<Func<IConnection>>();
```

Comment:
Should we make this an interface now to be clearer in DI? `IConnectionFactory`?

Reply:
Where would that interface live?